variables.tf

```diff
@@ -54,52 +54,48 @@ This object describes the public IP configuration when creating Nat Gateway's wi
 DESCRIPTION
 }
 
-# source https://github.com/Azure/terraform-azurerm-avm-res-network-virtualnetwork/blob/main/variables.tf#L306
 variable "subnets" {
   type = map(object({
-    address_prefix   = optional(string)
-    address_prefixes = optional(list(string))
-    name             = optional(string)
+    address_prefix                  = optional(string)
+    address_prefixes                = optional(list(string))
+    default_outbound_access_enabled = optional(bool, false)
+    delegate_to                     = optional(string, null)
     nat_gateway = optional(object({
       id = string
     }))
-    network_security_group = optional(object({
-      id = string
-    }))
+    name                          = optional(string)

```

Comment:
Nitpick: I see this in the other variables as well, wondering if we should just move the name always up to the top